packages/perseus-core/src/parse-perseus-json/general-purpose-parsers/object.ts

```diff
@@ -2,13 +2,20 @@ import {failure, isSuccess} from "../result";
 
 import {isObject} from "./is-object";
 
+import type {OptionalizeProperties} from "./object-types";
 import type {Mismatch, ParsedValue, Parser} from "../parser-types";
 
 type ObjectSchema = Record<keyof any, Parser<any>>;
 
-export function object<S extends ObjectSchema>(
+export function objectWithAllPropertiesRequired<S extends ObjectSchema>(
     schema: S,
 ): Parser<{[K in keyof S]: ParsedValue<S[K]>}> {
+    return object(schema) as any;
```

Comment:
This seems weird and makes me a little nervous, but I think it's safe. By equating `objectWithAllPropertiesRequired` and `object`, we are essentially saying that we don't care about the difference between `{foo: string | undefined}` and `{foo?: string}`. The difference according to TS is that `{foo: string | undefined}` requires `foo` to be present, so `{}` is not assignable to it.